src/App.js

```diff
@@ -62,33 +63,37 @@ export default function App() {
 
   return (
     <Appcontainer>
-      {isHidden || (
+      {currentPage === 'activity' && (
```

Comment:
You have three different components which you render when `currentPage === 'activity'`. Why not extract these to an extra `ActivityPage.js` file and move them there? That way your App.js would be a bit leaner and cleaner :)